### PR TITLE
fix: add pod namespace in node hostname

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

include pod namespace in Konnect node hostname for KIC nodes, using `namespace/name` pattern. If namespace or name is not found, fall back to use `os.Hostname`, which will be the pod name.

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/manager/run.go [hunk 1]
```diff
@@ -197,7 +197,16 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
...
 		} else {
-			hostname, _ := os.Hostname()
+			var hostname string
+			podName := os.Getenv("POD_NAME")
+			podNs := os.Getenv("POD_NAMESPACE")
+			if podName == "" || podNs == "" {
+				setupLog.Info("pod namespace or pod name not found, fallback to use hostname as node name in konnect")
+				hostname, _ = os.Hostname()
+			} else {
+				hostname = podNs + "/" + podName
```
Comment (pmalek) [inline internal/manager/run.go:207]: If `podNs` ends up empty then `hostname` becomes: `"/" + podName` and vice versa for empty `podName`. Is that intended?

File: internal/manager/run.go [hunk 2]
```diff
@@ -197,7 +197,16 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
...
+				setupLog.Info(fmt.Sprintf("use %s/%s as node name in konnect", podNs, podName))
```
Comment (pmalek) [inline internal/manager/run.go:208]:
```suggestion
				setupLog.Info(fmt.Sprintf("use %s as node name in konnect", hostname))
```

**Comment by pmalek** [comment]
We still didn't add a commit style enforcing tool :)

Please adjust the commit message to follow [conventional commits style guide](https://www.conventionalcommits.org/en/v1.0.0/#summary)

File: internal/manager/run.go (hunk 1 above)
Comment (randmonkey) [reply internal/manager/run.go:207]: I think either being empty will make the hostname be `os.Hostname()`.

**Comment by randmonkey** [comment]
> We still didn't add a commit style enforcing tool :)
> 
> Please adjust the commit message to follow [conventional commits style guide](https://www.conventionalcommits.org/en/v1.0.0/#summary)

changed the commit message.

File: internal/manager/run.go (hunk 2 above)
Comment (randmonkey) [reply internal/manager/run.go]: updated.

**Review by pmalek — APPROVED** [review]

**Pull request merged**